### ISSUE TITLE
Fails at choo's server side render

### ISSUE THREAD

**Issue by graforlock** [yoshuawuyts/microcomponent#20]
In the express environment the attempt of SSR rendering choo results in:

```this._element.getAttribute is not a function```

Workaround involves passing a mock function in the server side js context and importing the proper module from the client bundle endpoint. But its a hacky solution.

So, with code:
```javascript
`<div>${geoMap.render({ longitude, latitude, markers, city }, emit)}</div>`
```

One has to pass down  component in nodejs state environment as something like:

```javascript
export default {
 render: () => ''
};
```

I believe its not a big fix to make this assumption dependant on the runtime context.

I realise this is probably more to do with `nanocomponent` but it fails in both places and I only wanted to create one issue ticket.

**Comment by yoshuawuyts** [comment]
Ah, I was wondering if this would work - we should add some tests to both
{micro,nano}component

**Comment by graforlock** [comment]
Ok, so it fails due to microcomponent  wanting to get `.attributes` on the server (which isn't available):

https://github.com/yoshuawuyts/microcomponent/blob/master/index.js#L46

the fix is to replace the line with:

```javascript
if (oldElement && oldElement.attributes) {
```

It then renders fine on server side.

**Comment by yoshuawuyts** [comment]
Hey, so this module is mostly deprecated in favor of nanocomponent -
there's also funcomponent and component-box that are making some cool
progress and similar to this. If you're interested in this, consider
dropping by on IRC sometime! :D

**Comment by graforlock** [comment]
Hmm okay. I did a cleanup as  some folks are still using it and don't know it (including me (: ). Its closer to react mindset and some companies have mostly react devs that think and dream of this.props.

**Comment by yoshuawuyts** [comment]
Yeah, we definitely want to get closer to that. Just not entirely settled
on the approach yet; tho I feel like we can figure it out! :D
